src/darwin/Framework/CHIPTests/MTRDeviceTests.m

```diff
@@ -3020,6 +3021,120 @@ - (void)test032_MTRPathClassesEncoding
     XCTAssertEqualObjects(originalCommandPath, decodedCommandPath);
 }
 
+#ifdef DEBUG
+- (void)test033_TestMTRDeviceDeviceConfigurationChanged
+{
+    // Ensure the test starts with clean slate.
+    [sController.controllerDataStore clearAllStoredClusterData];
+    NSDictionary * storedClusterDataAfterClear = [sController.controllerDataStore getStoredClusterDataForNodeID:@(kDeviceId)];
+    XCTAssertEqual(storedClusterDataAfterClear.count, 0);
+
+    __auto_type * device = [MTRDevice deviceWithNodeID:kDeviceId deviceController:sController];
+    dispatch_queue_t queue = dispatch_get_main_queue();
+
+    // Check if subscription is set up and initial reports are received.
+    XCTestExpectation * subscriptionExpectation = [self expectationWithDescription:@"Subscription has been set up"];
+    XCTestExpectation * gotInitialReportsExpectation = [self expectationWithDescription:@"Initial Attribute and Event reports have been received"];
+
+    __auto_type * delegate = [[MTRDeviceTestDelegate alloc] init];
+    delegate.onReachable = ^() {
+        [subscriptionExpectation fulfill];
+    };
+
+    delegate.onReportEnd = ^() {
+        [gotInitialReportsExpectation fulfill];
+    };
+
+    [device setDelegate:delegate queue:queue];
+
+    // Wait for subscription set up and intitial reports received.
+    [self waitForExpectations:@[
+        subscriptionExpectation,
+        gotInitialReportsExpectation,
+    ]
+                      timeout:60];
+
+    XCTestExpectation * gotAttributeReportExpectation = [self expectationWithDescription:@"Attribute report has been received"];
+    XCTestExpectation * gotAttributeReportEndExpectation = [self expectationWithDescription:@"Attribute report has ended"];
+    XCTestExpectation * deviceConfigurationChangedExpectation = [self expectationWithDescription:@"Device configuration changed was receieved"];
+    __block unsigned attributeReportsReceived = 0;
+    delegate.onAttributeDataReceived = ^(NSArray<NSDictionary<NSString *, id> *> * attributeReport) {
+        attributeReportsReceived += attributeReport.count;
+        XCTAssert(attributeReportsReceived > 0);
+        for (NSDictionary<NSString *, id> * attributeDict in attributeReport) {
+            MTRAttributePath * attributePath = attributeDict[MTRAttributePathKey];
+            XCTAssert(attributePath != nil);
```

Comment:
```suggestion
            XCTAssertNotNil(attributePath);
```
and similar things elsewhere: use `XCTAssertEqual()`, `XCTAssertEqualObjects`, etc.  Things like: `XCTAssertEqualObjects(attributePath.cluster, @(MTRClusterDescriptorID)`.

Except `MTRClusterDescriptorID` is deprecated, so don't use that.